File: app/src/main/java/com/jakebarnby/filemanager/models/OneDriveFile.java
package com.jakebarnby.filemanager.models;

import android.net.Uri;

import com.microsoft.graph.extensions.DriveItem;

/**
 * Created by Jake on 6/7/2017.
 */

public class OneDriveFile extends SourceFile {
    public void setFileProperties(DriveItem file) {
        setUri(Uri.parse(file.webUrl));
        setName(file.name);
        setCanRead(true);
        setDirectory(file.folder != null);
    }
}

File: app/src/main/java/com/jakebarnby/filemanager/models/sources/onedrive/OneDriveFile.kt
package com.jakebarnby.filemanager.models.sources.onedrive

import androidx.room.Entity
import com.jakebarnby.filemanager.models.SourceFile
import com.jakebarnby.filemanager.models.SourceType
import com.microsoft.graph.extensions.DriveItem

/**
 * Created by Jake on 6/7/2017.
 */
class OneDriveFile(file: DriveItem) : SourceFile() {

    init {
        path = file.webUrl
        id = file.id.hashCode().toLong()
        parentFileId = file.parentReference?.id?.toLong() ?: -1L
        remoteFileId = file.id.hashCode().toLong()
        name = file.name
        sourceId = SourceType.ONEDRIVE.id
        isDirectory = file.folder != null
        size = file.size
        modifiedTime = file.lastModifiedDateTime.timeInMillis
        isHidden = file.name.startsWith(".")
        if (file.thumbnails != null && file.thumbnails.currentPage.size > 0) {
            thumbnailLink = file.thumbnails.currentPage[0].small.url
        }
    }
}